dhis-2/dhis-web/dhis-web-dataentry/src/main/java/org/hisp/dhis/de/action/LoadFormAction.java

```diff
@@ -481,11 +488,7 @@ private void getSectionForm( Collection<DataElement> dataElements, DataSet dataS
             categoryCombos.add( dataElement.getDataElementCategoryCombo( dataSet ) );
         }
 
-        List<CategoryCombo> listCategoryCombos = new ArrayList<>( categoryCombos );
-
-        Collections.sort( listCategoryCombos, new CategoryComboSizeComparator() );
-
-        return listCategoryCombos;
+        return categoryCombos.stream().sorted( new CategoryComboSizeNameComparator() ).collect( Collectors.toList() );
```

Comment:
I think now that you moved the List variable up, let's just use traditional sorting, less overhead and easier to read:

Collections.sort( categoryCombos, new CategoryComboSizeComparator() );